src/pages/Reports/ReportsTable.tsx

```diff
@@ -0,0 +1,83 @@
+import { AoGateway } from '@ar.io/sdk';
+import TableView from '@src/components/TableView';
+import useReports, { ReportTransactionData } from '@src/hooks/useReports';
+import { formatDate } from '@src/utils';
+import { ColumnDef, createColumnHelper } from '@tanstack/react-table';
+import { useNavigate } from 'react-router-dom';
+
+const columnHelper = createColumnHelper<ReportTransactionData>();
+
+const ReportsTable = ({
+  ownerId,
+  gateway,
+}: {
+  ownerId: string;
+  gateway: AoGateway;
+}) => {
+  const navigate = useNavigate();
+
+  const {
+    isLoading,
+    data: reports,
+    hasNextPage,
+    fetchNextPage,
+  } = useReports(ownerId, gateway);
+
+  // Define columns for the table
+  const columns: ColumnDef<ReportTransactionData, any>[] = [
```

Comment:
Alot of these column defs in various tables repeat the same info, each one defines an id, header, and accessor index. I would consider a generator function for these for templating some default values that take in a callback for manipulating individual columns you generate.

Reply:
I think the code is clear as-is as a pattern. We might save some lines of code but at the expense of clarity, and this follows common practice for tanstack tables so others using tanstack should be able to come in and make changes easily.